blockchain/api.go

```diff
@@ -1369,7 +1518,195 @@ func (api *BasicSimpleGatekeeper) Payout(ctx context.Context, key *ecdsa.Private
+		return err
+	}
+
+	return nil
+}
+
+func (api *BasicMultiSigAPI) ConfirmTransaction(ctx context.Context, key *ecdsa.PrivateKey, transactionId *big.Int) error {
+	opts := api.opts.getTxOpts(ctx, key, api.opts.gasLimit)
+	tx, err := api.contract.ConfirmTransaction(opts, transactionId)
+	if err != nil {
+		return err
+	}
+
+	if _, err := WaitTxAndExtractLog(ctx, api.client, api.opts.blockConfirmations, api.opts.logParsePeriod, tx, ConfirmationTopic); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (api *BasicMultiSigAPI) RevokeConfirmation(ctx context.Context, key *ecdsa.PrivateKey, transactionId *big.Int) error {
+	opts := api.opts.getTxOpts(ctx, key, api.opts.gasLimit)
+	tx, err := api.contract.RevokeConfirmation(opts, transactionId)
+	if err != nil {
+		return err
+	}
+
+	if _, err := WaitTxAndExtractLog(ctx, api.client, api.opts.blockConfirmations, api.opts.logParsePeriod, tx, RevocationTopic); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (api *BasicMultiSigAPI) ExecuteTransaction(ctx context.Context, key *ecdsa.PrivateKey, transactionId *big.Int) error {
+	opts := api.opts.getTxOpts(ctx, key, api.opts.gasLimit)
+	tx, err := api.contract.ExecuteTransaction(opts, transactionId)
+	if err != nil {
+		return err
+	}
+
+	if _, err := WaitTxAndExtractLog(ctx, api.client, api.opts.blockConfirmations, api.opts.logParsePeriod, tx, PayInTopic); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (api *BasicMultiSigAPI) IsConfirmed(ctx context.Context, transactionId *big.Int) (bool, error) {
+	return api.contract.IsConfirmed(getCallOptions(ctx), transactionId)
+}
+
+func (api *BasicMultiSigAPI) GetConfirmationCount(ctx context.Context, transactionId *big.Int) (*big.Int, error) {
+	return api.contract.GetConfirmationCount(getCallOptions(ctx), transactionId)
+}
+
+func (api *BasicMultiSigAPI) GetTransactionCount(ctx context.Context, penging bool, executed bool) (*big.Int, error) {
+	return api.contract.GetTransactionCount(getCallOptions(ctx), penging, executed)
+}
+
+func (api *BasicMultiSigAPI) GetOwners(ctx context.Context) ([]common.Address, error) {
+	return api.contract.GetOwners(getCallOptions(ctx))
+}
+
+func (api *BasicMultiSigAPI) GetConfirmations(ctx context.Context, transactionId *big.Int) ([]common.Address, error) {
+	return api.contract.GetConfirmations(getCallOptions(ctx), transactionId)
+}
+
+func (api *BasicMultiSigAPI) GetTransactionIds(ctx context.Context, from *big.Int, to *big.Int, pending bool, executed bool) ([]*big.Int, error) {
+	return api.contract.GetTransactionIds(getCallOptions(ctx), from, to, pending, executed)
+}
+
+func (api *BasicMultiSigAPI) GetTransaction(ctx context.Context, transactionId *big.Int) (*MultiSigTransactionData, error) {
+	tx, err := api.contract.Transactions(getCallOptions(ctx), transactionId)
+	if err != nil {
+		return nil, err
+	}
+	return &MultiSigTransactionData{
```

Comment:
that the reason for introducing new struct and not using `types.Transaction`?

Reply:
its not just blockchain tx

Reply:
but it looks like just a subset of fields from `types.Transaction`, isn't it?